### ISSUE TITLE
`client.disconnect()` should be blocked until unfulfilled tasks are complete

### ISSUE THREAD

**Issue by wilsonpage** [gaia-components/threads#29]

**Comment by wilsonpage** [comment]
@azasypkin is this what you want? Is this what that plugin you wrote does?

**Comment by azasypkin** [comment]
@wilsonpage yeah, somewhat similar, I wanted to have this for `client.destroy()` [1]

[1] https://github.com/mozilla-b2g/gaia/pull/31820/files#diff-fcaa28ac54966be861f211e9547ede17

**Comment by azasypkin** [comment]
But I'm not sure if it can be done in the bridge core since we should support streams that are in a separate plugin... Only if it's done on the lower level so that every plugin that uses `send` or `receive` is counted.
